app/redux/shared_actions/WorkoutActionCreators.js

```diff
@@ -25,3 +55,21 @@ export const endWorkout = () => (dispatch, getState) => {
 };
 
 export const autoEndWorkout = () => ({ type: END_WORKOUT });
+
+const logAnalytics = (num_sets, num_reps, num_removes, num_restores, num_sets_with_fields, state) => {
```

Comment:
instead of passing params in, IMO pass in the state and have this function pull what it needs for the analytics

Reply:
Make all of these state?